src/DatabaseSchema/TablesList.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Quidco\DbSampler\DatabaseSchema;
+
+use Quidco\DbSampler\Configuration\MigrationConfiguration;
+use Quidco\DbSampler\ReferenceStore;
+use Quidco\DbSampler\SamplerInterface;
+use Quidco\DbSampler\SamplerMap;
+
+class TablesList
+{
+    private $tables = [];
+
+    /**
+     * @var ReferenceStore
+     */
+    private $referenceStore;
+    /**
+     * @var array
+     */
+    private $rawTables;
+
+    private function __construct(array $rawTables)
+    {
+        $this->referenceStore = new ReferenceStore();
+        $this->rawTables = $rawTables;
+    }
+
+    public static function fromConfig(MigrationConfiguration $configuration): self
+    {
+        return new self((array)$configuration->getTables());
+    }
+
+    public function getTables(): array
+    {
+        // @todo we probably shouldn't be building the sampler in this getter. find another place to do it!
+        if ([] === $this->tables) {
+            foreach ($this->rawTables as $table => $migrationSpec) {
+                $this->tables[$table] = $this->buildTableSampler($migrationSpec);
+            }
+        }
+
+        return $this->tables;
+    }
+
+    /**
+     * Build a SamplerInterface object from configuration
+     *
+     * @throws \UnexpectedValueException If bad object created - should be impossible
+     * @throws \RuntimeException On invalid specification
+     */
+    private function buildTableSampler(\stdClass $migrationSpec): SamplerInterface
+    {
+        $sampler = null;
+
+        // @todo: $migrationSpec should be an object with a getSampler() method
+        $samplerType = strtolower($migrationSpec->sampler);
+        if (array_key_exists($samplerType, SamplerMap::MAP)) {
+            $samplerClass = SamplerMap::MAP[$samplerType];
+            $sampler = new $samplerClass;
+            if (!$sampler instanceof SamplerInterface) {
+                throw new \UnexpectedValueException('Invalid sampler created');
+            }
+            /** @var SamplerInterface $sampler */
+            $sampler->loadConfig($migrationSpec);
+            $sampler->setReferenceStore($this->referenceStore);
+        } else {
+            throw new \RuntimeException("Unrecognised sampler type '$samplerType' required");
+        }
+
+        return $sampler;
```

Comment:
I'm sorry I only saw this now. Worth changing?

```suggestion
if (array_key_exists(...)) {
    $sampler = new SamplerMap::MAP[...]();
...
    return $sampler;
}

throw new \RuntimeException('.....');
```

The way it is currently it'll always throw if the key doesn't exist, so this way makes more sense to me. Means you could probably drop the $sampler = null; on L54 as well

Reply:
Does make for a neater implementation.

Reply:
Very happy for this to be deferred to a later date